Fix Qt5.10.1 to build against newer glibc.
This takes the relevant part of patch from https://code.qt.io/cgit/qt/qtbase.git/commit/?id=25feee4e061b99edab79503d81f5bd045c6c8e3d
Added to EasyBuild by Simon Branford (University of Birmingham)
index 3469ebe5e6..c5585ea32a 100644
--- a/qtbase/src/corelib/global/qconfig-bootstrapped.h
+++ b/qtbase/src/corelib/global/qconfig-bootstrapped.h
@@ -98,9 +98,17 @@
 #define QT_NO_QOBJECT
 #define QT_FEATURE_process -1
-#define QT_FEATURE_renameat2 -1
+#ifdef __GLIBC_PREREQ
+# define QT_FEATURE_renameat2 (__GLIBC_PREREQ(2, 28) ? 1 : -1)
+#else
+# define QT_FEATURE_renameat2 -1
+#endif
 #define QT_FEATURE_sharedmemory -1
 #define QT_FEATURE_slog2 -1
-#define QT_FEATURE_statx -1
+#ifdef __GLIBC_PREREQ
+# define QT_FEATURE_statx (__GLIBC_PREREQ(2, 28) ? 1 : -1)
+#else
+# define QT_FEATURE_statx -1
+#endif
 #define QT_FEATURE_syslog -1
 #define QT_NO_SYSTEMLOCALE
 #define QT_FEATURE_systemsemaphore -1
